osquery/core/system.cpp

```diff
@@ -131,7 +131,13 @@ Status checkStalePid(const std::string& content) {
       return Status(1, "Error querying processes: " + q.getMessageString());
     }
 
-    if (q.rows().size() >= 1 && q.rows().front()["name"] == "osqueryd") {
+    bool is_osqueryd = false;
+    if (q.rows().size() >= 1) {
```

Comment:
Actually, this makes me realize that the query should just be checking for the exact condition we want. Why do this all ourselves when SQLite does it for us?

Query changes to
```
query_text << "SELECT name FROM processes WHERE pid = " << pid << " AND name = 'osqueryd';";
```
and then we just check if there's a row.